Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.12.5-alpine3.9 as build
+FROM golang:1.16-alpine as build
```

Comment:
```suggestion
FROM golang:1.16.15-alpine3.15 as build
```

Is there a reason you're using a looser pinning?

Reply:
I just picked one at random. I doubt they will change after 2 years but its possible. I don't really know how they decide to tag images. 
Should we do the following instead?
```
FROM golang@sha256:9743f230f26d1e300545f0330fd4a514f554c535d967563ee77bf634906502b6 as build
# This golang:1.16-alpine3.15 was tagged Mar 3, 2022 at 9:02 pm
```

Reply:
I doubt it too but explicit is better than implicit. I think the version tags are good enough and more readable than a hash